applications/OptimizationApplication/python_scripts/controls/material/material_properties_control.py

```diff
@@ -0,0 +1,102 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.controls.control import Control
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.helper_utilities import IsSameContainerExpression
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+    return MaterialPropertiesControl(model, parameters, optimization_problem)
+
+class MaterialPropertiesControl(Control):
+    """Material properties control
+
+    This is a generic material properties control which creates a control
+    for the specified control variable. This does not do any filtering.
+
+    TODO: Extend with filtering techniques when they are implemented.
+
+    """
+    def __init__(self, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__()
+
+        default_settings = Kratos.Parameters("""{
+            "combined_output_model_part_name": "<CONTROL_NAME>_combined_no_neighbours_control",
+            "model_part_names"               : [""],
+            "control_variable_name"          : ""
+        }""")
+        parameters.ValidateAndAssignDefaults(default_settings)
+
+        self.output_model_part_name = parameters["combined_output_model_part_name"].GetString()
+        self.model_part_names = parameters["model_part_names"].GetStringArray()
+        self.optimization_problem = optimization_problem
```

Comment:
I would put self.optimization_problem = optimization_problem and self.model = model at the base class otherwise it gets repeated ?!

Reply:
I would keep the repition. I like to keep the interfaces fully clean. because, some controls may do everything in the `__init__` and then there is no point having them in the base class.